### PR TITLE
Adds option to disable promotions

### LINKED ISSUE #12313: Option to disable promotions
## Test plan

https://github.com/brave/browser-laptop/pull/12314#issue-282850907

----

### Description

We need to add a mechanism to disable promotions, so that users don't see them if they never enable wallet. We don't want to call server in this cases.

based on the conversation with @BrendanEich, suggestion is to add toggle into advance settings. This way user can disable promotions without going into the payments and enabling wallet.

cc @bradleyrichter @davidtemkin

### PR DESCRIPTION
**Pull request by NejcZdovc**
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Resolves #12313

Auditors:

## Test Plan:
- clean profile
- start brave with LEDGER_VERBOSE
- as soon as brave starts go to preferences->advance settings and disable promotions (you have 15s to do this)
- there shouldn't be any requests to get promotions (like `https://ledger.mercury.basicattentiontoken.org/v1/grants`)

## Reviewer Checklist:

Tests

- [x] Adequate test coverage exists to prevent regressions
- [x] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [x] New files have MPL2 license header

### CODE REVIEW

**Comment by NejcZdovc** [comment]
this is how options looks like

image

File: app/extensions/brave/locales/en-US/preferences.properties [hunk 1]
```diff
@@ -215,6 +215,7 @@ pageNofMText=Page {{n}} of {{m}}
 paintTabs=Show tabs in page theme color
 passwordManager=Password Manager
 passwordsAndForms=Passwords and Forms
+paymentsAllowPromotions=Show BAT promotions
```
Comment (bsclifton) [inline app/extensions/brave/locales/en-US/preferences.properties]: Per comment from @davidtemkin, I believe this should be "token" promotions

Comment (NejcZdovc) [reply app/extensions/brave/locales/en-US/preferences.properties]: done

Comment (davidtemkin) [reply app/extensions/brave/locales/en-US/preferences.properties]: yes, thanks

**Comment by bsclifton** [comment]
master 4028c6617c9d7194c337ea088871cd36f90bcefc
0.21.x e84a17ed68bdc307c1c129cb71aae8f6db99402b
0.20.x 8b1a5a89cb296d430b7c86601e8a7a82faac80a2

**Pull request merged**